src/background/translate-persisted-data.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { PersistedData } from 'background/get-persisted-data';
+import {
+    InjectingState,
+    VisualizationStoreData,
+} from 'common/types/store-data/visualization-store-data';
+import { forEach } from 'lodash';
+
+export function translatePersistedData(persistedData: PersistedData): PersistedData {
+    forEach(persistedData.tabData, tabData => {
+        translateInjectionState(tabData.visualizationStoreData);
+    });
+
+    return persistedData;
+}
+
+function translateInjectionState(data: VisualizationStoreData) {
```

Comment:
There's already a layer that exists to handle merging in old persisted state as a new initial store state (`initial-visualization-store-data.ts`) - we should reuse that instead of creating a new one.

Reply:
Fixed, thanks!